### DISCUSSION TITLE
Edit Photo end Caption end ReplyMarkup

### DISCUSSION THREAD

**Discussion by Godbalance** [mymmrac/telego#89] [Q&A] [answered]
I don't understand how to change the photo and description.  I don't want to send a new photo with a description.  I want to change what has already been submitted.  But I didn't find an example.  Help.

**Comment by mymmrac** [comment, marked as answer]
Hello, thanks for question, I added example [here](https://github.com/mymmrac/telego/blob/main/examples/edit_message/main.go).

In short, you need to use `bot.EditMessageCaption` to edit message text and `bot.EditMessageMedia` to change photo in message.

You also need to know message ID, you can use direct `MessageID` retrieved from message sent or from some update, or you can use `InlineMessageID` from callback query update that is sent when user clicks on inline button.

**Comment by Godbalance** [comment]
Thank you very much.  But I don't understand how to get InlineMessageID.

**Reply by mymmrac** [reply]
When you are handling updates, in callback query there is exact field you need, please refer: https://core.telegram.org/bots/api#callbackquery (field `InlineMessageID`)

**Reply by mymmrac** [reply]
Or just use `Message` field, that has ID

**Reply by mymmrac** [reply]
Using bit handlers you can do that like this:
```go
bh.HandleCallbackQuery(func(bot *telego.Bot, query telego.CallbackQuery) {
	_, _ = bot.EditMessageCaption(&telego.EditMessageCaptionParams{
		ChatID:    tu.ID(query.Message.Chat.ID),
		MessageID: query.Message.MessageID,
		Caption:   "New caption",
	})

	_ = bot.AnswerCallbackQuery(tu.CallbackQuery(query.ID).WithText("Updated"))
}, th.AnyCallbackQueryWithMessage())
```

> Note: I strongly recommend handling error which are ignored here
